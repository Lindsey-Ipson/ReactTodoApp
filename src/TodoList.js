import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, newTodo]);
  }

  const remove = (id) => {
    setTodos((todos) => todos.filter(todo => todo.id !== id));
  }

  return (
    <div className="TodoList">
      <NewTodoForm addTodo={addTodo} />
      {todos.map(todo => (
        <Todo
          key={todo.id}
          id={todo.id}
          task={todo.task}
          remove={remove}
        />
      ))}
    </div>
  )

}

export default TodoList;