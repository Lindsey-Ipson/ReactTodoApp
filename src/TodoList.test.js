import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function submitTodoHelper(todoList, task = "New task") {
  const taskInput = todoList.getByLabelText("Task");
  const submitButton = todoList.getByText("Add todo");

  fireEvent.change(taskInput, { target: { value: task } });
  fireEvent.click(submitButton);
}

test("renders without crashing", function() {
  render(<TodoList />);
});

test("matches snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("adds a new todo", function() {
  const todoList = render(<TodoList />);

  expect(todoList.queryByText("New task")).not.toBeInTheDocument();

  submitTodoHelper(todoList);

  // expect the form to be reset to empty and new todo to be displayed
  expect(todoList.getByLabelText("Task")).toHaveValue("");
  expect(todoList.getByText("New task")).toBeInTheDocument();
  expect(todoList.getByText("X")).toBeInTheDocument();
});

test("deletes a todo", function() {
  const todoList = render(<TodoList />);

  submitTodoHelper(todoList);

  expect(todoList.queryByText("New task")).toBeInTheDocument();

  const deleteButton = todoList.getByText("X");
  fireEvent.click(deleteButton);

  expect(todoList.queryByText("New task")).not.toBeInTheDocument();
});

