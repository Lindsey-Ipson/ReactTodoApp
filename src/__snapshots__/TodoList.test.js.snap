// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form
      class="NewTodoForm"
    >
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
